ui/StatusQ/src/StatusQ/Components/StatusCard.qml

```diff
@@ -162,126 +181,178 @@ Rectangle {
+            input.rightComponent: Row {
+                width: implicitWidth
+                spacing: 4
+                visible: root.state !== "error"
+                StatusFlatRoundButton {
+                    anchors.verticalCenter: parent.verticalCenter
+                    width: 12
+                    height: 12
+                    icon.name: root.locked && advancedInput.tempLock ? "lock" : "unlock"
+                    icon.width: 12
+                    icon.height: 12
+                    icon.color: root.locked && advancedInput.tempLock ? Theme.palette.primaryColor1 : Theme.palette.baseColor1
+                    type: StatusFlatRoundButton.Type.Secondary
+                    enabled: !disabled
+                    onClicked: {
+                        advancedInput.tempLock = !advancedInput.tempLock
+                        root.locked = advancedInput.tempLock
+                        root.cardLocked(advancedInput.tempLock)
                     }
                 }
-                text: root.preCalculatedAdvancedText
-                onTextChanged: {
-                    advancedInput.tempLock = false
-                    waitTimer.restart()
+            }
+            validators: [
+                StatusFloatValidator {
+                    id: floatValidator
+                    bottom: 0
+                    top: root.maxAdvancedValue
+                    errorMessage: ""
                 }
-                Timer {
-                    id: waitTimer
-                    interval: lockTimeout
-                    onTriggered: {
-                        advancedInput.tempLock = true
-                        if(!!advancedInput.text && root.preCalculatedAdvancedText !== advancedInput.text) {
-                            root.cardLocked(advancedInput.tempLock)
-                        }
+            ]
+            text: root.preCalculatedAdvancedText
+            onTextChanged: {
+                advancedInput.tempLock = false
+                waitTimer.restart()
+            }
+            Timer {
+                id: waitTimer
+                interval: lockTimeout
+                onTriggered: {
+                    advancedInput.tempLock = true
+                    if(!!advancedInput.text && root.preCalculatedAdvancedText !== advancedInput.text) {
+                        root.locked = advancedInput.tempLock
+                        root.cardLocked(advancedInput.tempLock)
                     }
                 }
             }
-            StatusBaseText {
-                id: tertiaryText
-                Layout.maximumWidth: root.width - 12
-                elide: Text.ElideRight
-                font.pixelSize: 10
-            }
         }
-        StatusIcon {
-            id: cardIcon
-            Layout.alignment: Qt.AlignTop | Qt.AlignRight
-            Layout.preferredHeight: 32
-            Layout.preferredWidth: 32
-            mipmap: true
+        StatusBaseText {
+            id: tertiaryText
```

Comment:
I find having the same name as the `root.tertiaryText` property confusing. How about `footerText`?